### PR TITLE
[Docs] Add description about how RGP evaluation works

### PR DESCRIPTION
**Pull request by yhyakuna**
This PR is related to the [customer issue with RGP](https://docs.google.com/document/d/1-Z0LPyvtXAMo3zcGHP6S63WDFKa065Ng9swjqMHt_nM/edit?usp=sharing).

🎫 [Jira: SPE-299](https://hashicorp.atlassian.net/browse/SPE-299)

🔍 [Deploy preview](https://vault-git-docs-rgp-and-namespaces-hashicorp.vercel.app/vault/docs/enterprise/sentinel)

This PR:
- Adds the enforcement level info: advisory, soft-mandatory, and hard-mandatory
- Visual diagram to explain the policy evaluation workflow
- Add the "RGPs and namespaces" section

### CODE REVIEW

File: website/content/docs/enterprise/sentinel/index.mdx [hunk 1]
```diff
@@ -6,7 +6,13 @@ description: An overview of how Sentinel interacts with Vault Enterprise.
...
 # Overview
 
--> **Note**: This feature requires [Vault Enterprise Plus](https://www.hashicorp.com/products/vault/).
+<Note>
+
+Sentinel requires [Vault Enterprise
+Plus](https://www.hashicorp.com/products/vault/pricing/) license, or an [HCP
+Vault Plus](/hcp/docs/vault/tiers-and-features#plus-tier) cluster.
+
+</Note>
```
Comment (maxb) [inline website/content/docs/enterprise/sentinel/index.mdx:15]: Whilst this block is being updated, would you be open to taking a look at https://github.com/hashicorp/vault/issues/18525, which I filed many months ago, calling out inconsistencies in the self-hosted Enterprise tier in various parts of the docs, including this one?

File: website/content/docs/enterprise/sentinel/index.mdx [hunk 2]
```diff
@@ -42,22 +48,89 @@ Sentinel execution should be considered to be significantly slower than normal
...
-   granted capabilities on a path via ACL policies denies the request.
-2. RGPs attached to the token are evaluated. All policies must pass according
+![Policy evaluation](/img/diagram-policy-evaluation-workflow_light.png#light-theme-only)
+![Policy evaluation](/img/diagram-policy-evaluation-workflow_dark.png#dark-theme-only)
+
+1. If the request is unauthenticated, skip to evaluating the EGPs. Otherwise,
+   evaluate the token's ACL policies. These must grant access; as always, a
+   failure to be granted capabilities on a path via ACL policies denies the
+   request.
+2. Evaluate RGPs attached to the cliet token. All policies must pass according
```
Comment (maxb) [inline website/content/docs/enterprise/sentinel/index.mdx:74]:
```suggestion
2. Evaluate RGPs attached to the client token. All policies must pass according
```

File: website/content/docs/enterprise/sentinel/index.mdx [hunk 3]
```diff
@@ -42,22 +48,89 @@ Sentinel execution should be considered to be significantly slower than normal
...
+The training namespace is a child namespace of the education namespace. The "Sun
+Shine" entity created in the education namespace is a member of the "Tester"
+group which is defined in the training namespace. The group members inherit the
+group-level policy.
+
+![Relationship](/img/diagram-rgp-namespace_light.png#light-theme-only)
+![Relationship](/img/diagram-rgp-namespace_dark.png#dark-theme-only)
+
+While ACL policies and EGPs set rules on a specific path, an RGP may not have a
+target path. RGPs are tied to tokens, identity entities, or identity groups that
```
Comment (maxb) [inline website/content/docs/enterprise/sentinel/index.mdx:108]: The phrasing "may not" is open to multiple interpretations - to a reader without existing subject matter expertise, they may be uncertain whether the "it is uncertain whether it does or not" meaning or "it is not permitted" meaning is intended. Using "does not" would avoid the potential ambiguity.

File: website/content/docs/enterprise/sentinel/index.mdx (hunk 1 above)
Comment (yhyakuna) [reply website/content/docs/enterprise/sentinel/index.mdx:15]: I created a [Jira ticket](https://hashicorp.atlassian.net/browse/SPE-322) to follow up on this.  The issue was assigned to someone who left the company that it was left opened. Sorry!

**Review by vishalnayak — APPROVED** [review]

**Pull request merged**